### PR TITLE
Improve plugin.php

### PR DESCRIPTION
**Pull request by szepeviktor**
Add some parts from https://github.com/szepeviktor/starter-plugin/blob/master/README.md#parts-of-plugin-main-file
There are many more possible parts to add.

A plugin should do things in time.
But I let you _take over WordPress_ and close this PR!

### CODE REVIEW

File: plugin.php [hunk 1]
```diff
@@ -17,11 +17,16 @@
  */
 
 // Pull in the autoloader
-require_once __DIR__.'/autoload.php';
+if (! class_exists(Cone\Plugin::class)) {
+    require_once __DIR__.'/autoload.php';
+}
```
Comment (szepeviktor) [inline plugin.php:22]: Support Composer managed WordPress installations.
This needs a `composer.json` with autoloading configured.

File: plugin.php [hunk 2]
```diff
@@ -17,11 +17,16 @@
...
 
 // Register the activation and the deactivation hooks
 register_activation_hook(__FILE__, [Cone\Plugin::class, 'activate']);
 register_deactivation_hook(__FILE__, [Cone\Plugin::class, 'deactivate']);
 
 // Boot the plugin
-Cone\Plugin::boot();
+add_action('plugins_loaded', [Cone\Plugin::class, 'boot'], 10, 0);
```
Comment (szepeviktor) [inline plugin.php:29]: A plugin should not do _tricky_ things before `plugins_loaded`.

File: plugin.php [hunk 3]
```diff
@@ -17,11 +17,16 @@
...
+
+// Load translations
+add_action('init', [Cone\Plugin::class, 'loadTextDomain'], 10, 0);
```
Comment (szepeviktor) [inline plugin.php:32]: Load translations as late as possible.
This references a non-existent method!

Comment (szepeviktor) [reply plugin.php:32]: This could be moved into boot method but that is twisted logic.

**Comment by iamgergo** [comment]
Merci!

**Pull request merged**